src/types.rs

```diff
@@ -24,5 +26,133 @@ pub struct TransactionInput {
 
 #[derive(Serialize, Deserialize)]
 pub struct Response {
-    pub method: Option<Function>,
+    pub method: Option<ResponseMethod>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ResponseMethod {
+    name: String,
+    inputs: Vec<ResponseParam>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ResponseParam {
+    name: String,
+    internal_type: Option<String>,
+    #[serde(rename(serialize = "type"))]
+    kind: String,
+    value: String,
+}
+
+trait ResponseDisplay {
+    fn display(&self) -> String;
+}
+
+impl ResponseDisplay for ParamType {
+    fn display(&self) -> String {
+        match self {
+            ParamType::String => "string".to_string(),
+            ParamType::Address => "address".to_string(),
+            ParamType::Bytes => "bytes".to_string(),
+            ParamType::Int(n) => format!("int{}", n),
+            ParamType::Uint(n) => format!("uint{}", n),
+            ParamType::Bool => "bool".to_string(),
+            ParamType::Array(t) => format!("{}[]", t.display()),
+            ParamType::FixedBytes(n) => format!("bytes{}", n),
+            ParamType::FixedArray(t, n) => format!("{}[{}]", t.display(), n),
+            ParamType::Tuple(_) => "tuple".to_string(),
+        }
+    }
+}
+
+impl ResponseDisplay for &Vec<u8> {
```

Comment:
it's better to use slices (`&[u8]`) instead of vecs whenever possible: 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4d8d8926140fce7752539a113fa2b050